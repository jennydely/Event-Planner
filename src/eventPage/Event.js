import React from 'react';
import styled from 'styled-components/macro'
import {formatDate} from '../utils/date'

export default function Event({event}) {
const{eventName, eventLocation, eventDate} = event

    return (
        <EventItem>
            <Title>{eventName} - {eventLocation}</Title>
            <Duration>{formatDate(eventDate)}</Duration>
        </EventItem>
    )
}



const EventItem = styled.li`
list-style:none;
border: 2px solid #964B00;
marign: 0;
margin-bottom:20px;
padding:4px;
text-align:center;
`
const Title = styled.h2`
margin: 0;
font-size: 100%;
`
const Duration = styled.p`
margin:0;
`