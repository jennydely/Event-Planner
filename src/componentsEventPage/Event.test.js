import React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Event from './Event'
import 'jest-styled-components'

describe('Event', () => {
    const name = 'This is the eventname'
    const location = 'This is the eventlocation'
    const date = '06/25/2020 - 06/28/2020'


    it('display the eventname and eventlocation', () => {
        const { getByText } = render(<Event event={{eventName: name, eventLocation: location, eventDate: date}} />)
        expect(getByText(name + ' - ' + location)).toBeInTheDocument()
        expect(getByText(date)).toBeInTheDocument()
    })
    it('renders correctly', () => {
        const tree = renderer.create(<Event event={{eventName: name, eventLocation: location, eventDate: date}}/>)
        expect(tree).toMatchSnapshot()
    })
 } )