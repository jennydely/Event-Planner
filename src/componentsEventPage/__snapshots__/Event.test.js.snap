// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event renders correctly 1`] = `
.c0 {
  list-style: none;
  border: 2px solid brown;
  margin: 20px;
  padding: 4px;
}

.c1 {
  margin: 0;
}

.c2 {
  margin: 0;
}

<li
  className="c0"
>
  <h2
    className="c1"
  >
    This is the eventname
     - 
    This is the eventlocation
  </h2>
  <p
    className="c2"
  >
    This is the eventdate
  </p>
</li>
`;
